### DISCUSSION TITLE
Only allow full locales for translations

### DISCUSSION THREAD

**Discussion by LarsDenBakker** [ing-bank/lion#187] [Ideas]
Moved here from an internally created issue.

Currently inside ing-web components there are switch statements like this:

```js
switch (locale) {
  case 'en-GB':
    return import('./en-GB.js');
  case 'en-US':
    return import('./en-US.js');
  case 'en': 
    return import('./en.js');
}
``` 

Where both language-only ('en') and locales ('en-GB', 'en-US') are supported.

I think we should not support only 'language' specifiers within the localization system for a few reasons:
- It adds extra boilerplate code, and adds extra lazy loaded chunks for tools to manage.
- It makes tools based off of translations harder to write (we have a tool for managing translations for copy writers and a webpack plugin for optimizing translation bundling)
- It picks an implicit locale for localizing numbers and dates, possibly creating unexpected results for the user and potentially different across browsers.

**Comment by bashmish** [comment]
We had some time to digest this already and collect some knowledge.

First of all, this is not a theoretical problem, we already saw bugs because of it, e.g. [this one](https://github.com/ing-bank/lion/issues/90).

Second, we found out that there is [another problem with Google Chrome translate feature](https://github.com/ing-bank/lion/issues/152) which makes it difficult to fully forbid the usage of only language specifiers. Google Chrome itself uses language only (you translate the page to English, Google Chrome sets `en`), thus interfering with our system and forcing us to support them to a certain degree, or at least not throw, because it interrupts the Google Chrome injected JS code.

I think the approach here should be practical: yes, we don't want app developers to use language only specifiers, because they will not work correctly with dates and numbers, but we don't want to break other tools like Google Chrome translate. And how to technically implement such a fix I'm not so sure yet. We can detect Google Chrome injected scripts on a page and then temporarily allow language specifiers, but this is a fragile hack on our side. And who knows how other tools/plugins are going to use `<html lang>`. Maybe we should just write a linter which will detect the wrong `lang` attribute and we can add a `console.warning` instead of an exception for developers to notice the problem in the runtime. And this will not be visible that much in the apps which follow the rule of using full locales, unless users activate Google Chrome translate on them.

What do you think?

**Comment by jorenbroekema** [comment]
Now that we no longer conflict with third party translation tools, I am not really sure what to do about this one. I think we will keep allowing language only locales just because of the fact that we want to support tools like Google Translate. @tlouisse  opinion?

**Comment by jorenbroekema** [comment]
I'm going to move this to discussions, as I would classify this as intended behaviour and not a bug. Feel free to let me know if you think it should be a bug and why.
